### ISSUE TITLE
Pylance in Base and Strict type checking mode cannot determine right property type

### ISSUE THREAD

**Issue by MaxLeidlTEXSIB** [microsoft/pyright#3762]
## Environment data

-   Language Server version: v2022.7.40
-   OS and version: Windows 11 and Debian GNU/Linux 11 (bullseye)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.2 (Windows) and 3.9.2 (Debian)

## Code Snippet

```python
class ClassWithProperty:
    @property
    def name(self) -> str:
        return "bar"


def function_that_takes_property(foo: property):
    print(foo)


function_that_takes_property(ClassWithProperty.name)

```

## Repro Steps

1. Make a function that has an argument of type property (or a subclass of property).
2. Call the function with a class's property.

## Expected behavior

No warning from pylance.

## Actual behavior

The code runs without issues, but pylance shows the following warning:

```
Argument of type "property" cannot be assigned to parameter "foo" of type "property" in function "function_that_takes_property"
  "main.ClassWithProperty.__property_name" is incompatible with "builtins.property"
```

## Logs

```
(3380) Background analysis message: getDiagnosticsForRange
(3380) Background analysis message: getDiagnosticsForRange
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.265, which I just published. It will be included in a future release of pylance.

**Comment by MaxLeidlTEXSIB** [comment]
I tested it, and it works for the property class, as in my example. Thanks :)
I also tested it with a class, that subclasses property and there it sadly doesn't work yet.

I altered the above code snippet to include a property subclass which is then used in the function instead.

```python
class PropertySubClass(property):
    ...


class ClassWithProperty:
    @PropertySubClass
    def name(self) -> str:
        return "bar"


def function_that_takes_property(foo: PropertySubClass):
    print(foo)


function_that_takes_property(ClassWithProperty.name)
```

The code runs just fine, but the following error is printed by pyright/Pylance: 
```
Argument of type "PropertySubClass" cannot be assigned to parameter "foo" of type "PropertySubClass" in function "function_that_takes_property"
  "main.ClassWithProperty.__property_name" is incompatible with "main.PropertySubClass"
```

**Comment by erictraut** [comment]
Yeah, that won't work. Type checkers like pyright contain a bunch of special-case logic to make `@property` work. If you want to create property-like classes, I recommend creating a class that implements the descriptor protocol directly rather than trying to subclass property.

**Comment by MaxLeidlTEXSIB** [comment]
Yeah, I figured that working with the default property is probably hard. I followed your advice and it works great now, thanks :)
